### ISSUE TITLE
logical/postgresq: Graceful revocation failure

### ISSUE THREAD

**Issue by BSick7** [hashicorp/vault#1633]
When trying to revoke postgresql credentials, I am getting the following error:

```
Unmount error: Error making API request.

URL: DELETE http://<scrubbed>:8200/v1/sys/mounts/<scrubbed>
Code: 400. Errors:

* failed to revoke '<scrubbed>/creds/app/44cc2719-1791-3d27-9fc3-3951cb71cc03' (24 / 281): failed to revoke entry: resp:(*logical.Response)(nil) err:could not perform all revocation statements: pq: role "token-<scrubbed>-066e2cd1-9656-570c-c81b-de271d85bd53" does not exist
```

This mount should look for "does not exist" error messages and gracefully move on.

**Comment by jefferai** [comment]
Keying on text in error messages is generally a bad idea. Postgres has standard error codes but I don't see one that applies in this situation (https://www.postgresql.org/docs/9.5/static/errcodes-appendix.html). Ideally a role not existing would be a specific error code so if something changes Vault doesn't stop working.

You can work around this easily within Vault by using `revoke-force` (https://www.vaultproject.io/docs/http/sys-revoke-force.html). On the CLI this corresponds to the `revoke` command with `prefix` and `force` set.

**Comment by BSick7** [comment]
> Keying on text in error messages is generally a bad idea.
I agree.

It looks like https://github.com/hashicorp/vault/commit/f9d40aa63be242789d2c6725e727b841bc3776ef will fix this.

It would be nice to include this in `BUG FIXES` section of the CHANGELOG.

**Comment by BSick7** [comment]
Thanks for the `revoke-force` tip.

**Comment by jefferai** [comment]
Sorry, how does f9d40aa fix this?

**Comment by BSick7** [comment]
See the following lines:
https://github.com/hashicorp/vault/blob/f9d40aa63be242789d2c6725e727b841bc3776ef/builtin/logical/postgresql/secret_creds.go#L100-L110

**Comment by jefferai** [comment]
Fixed by 46ecb9a
